### ISSUE TITLE
move machine learning features to trait

### ISSUE THREAD

**Issue by mxfactorial** [mxfactorial/geonum#47]
### current

machine learning features in ./src/geonum_mod.rs:

impls:
pub fn regression_from(cov_xy: f64, var_x: f64) -> Self
pub fn perceptron_update(&self, learning_rate: f64, error: f64, input: &Geonum) -> Self
pub fn forward_pass(&self, weight: &Geonum, bias: &Geonum) -> Self
pub fn activate(&self, activation: Activation) -> Self

enum:
Activation

tests:
it_computes_regression_from_covariance
it_updates_perceptron_weights
it_performs_neural_network_operations

### next

similar to ./src/traits/optics.rs

move machine learning features to new ./traits/ml.rs

ml in Cargo.toml features

ml in Cargo.toml "all" feature list

tests pass

### hourly estimate

1

**Comment by mxfactorial** [comment]
release/0.6.9

added

- machine learning trait module (src/traits/machine_learning.rs)
- new MachineLearning trait with o(1) complexity ml operations
- activation enum supporting relu, sigmoid, tanh, and identity functions
- regression_from() - creates regression lines from covariance/variance with direct angle encoding
- perceptron_update() - updates weights using angle-based learning rules
- forward_pass() - performs neural network forward passes via geometric products
- activate() - applies activation functions through geometric transformations
- new ml feature flag for conditional compilation
- comprehensive test suite with 3 test functions covering all ml operations

changed

- refactored ml functions from src/geonum_mod.rs into separate trait module
- updated Cargo.toml to include ml feature and add to all feature set

technical details

- maintains geometric algebras o(1) complexity advantage over traditional o(n²) and o(2^n) ml operations
- follows existing trait pattern used by optics, projection, and manifold modules
- all 196 tests pass including ml-specific functionality

**Comment by mxfactorial** [comment]
hours worked:
1
